### ISSUE TITLE
pretrain loss with negative value

### ISSUE THREAD

**Issue by zhihuiY** [HannesStark/self-supervised-graphs#18]
hello everyone! I need help. When I repretrain the model with the commend
`python train.py --config=configs_clean/pre-train_QM9.yml`

there is negative value starting in the second epoch, this is the loss in epoch 200 and the reult:
`[Epoch 199] NTXent: -1.781640 val loss: -1.781640
[Epoch 200; Iter     2/  100] train: loss: -2.2522569
[Epoch 200; Iter     4/  100] train: loss: -2.2169321
[Epoch 200; Iter     6/  100] train: loss: -2.2570577
[Epoch 200; Iter     8/  100] train: loss: -2.2113991
[Epoch 200; Iter    10/  100] train: loss: -2.2871132
[Epoch 200; Iter    12/  100] train: loss: -2.1956975
[Epoch 200; Iter    14/  100] train: loss: -2.2575712
[Epoch 200; Iter    16/  100] train: loss: -2.2456131
[Epoch 200; Iter    18/  100] train: loss: -2.2454598
[Epoch 200; Iter    20/  100] train: loss: -2.2255766
[Epoch 200; Iter    22/  100] train: loss: -2.2309322
[Epoch 200; Iter    24/  100] train: loss: -2.2573645
[Epoch 200; Iter    26/  100] train: loss: -2.2338004
[Epoch 200; Iter    28/  100] train: loss: -2.2719038
[Epoch 200; Iter    30/  100] train: loss: -2.1639729
[Epoch 200; Iter    32/  100] train: loss: -2.1668520
[Epoch 200; Iter    34/  100] train: loss: -2.2000210
[Epoch 200; Iter    36/  100] train: loss: -2.2143204
[Epoch 200; Iter    38/  100] train: loss: -2.1709681
[Epoch 200; Iter    40/  100] train: loss: -2.1966450
[Epoch 200; Iter    42/  100] train: loss: -2.2163792
[Epoch 200; Iter    44/  100] train: loss: -2.2385902
[Epoch 200; Iter    46/  100] train: loss: -2.2735734
[Epoch 200; Iter    48/  100] train: loss: -2.2653208
[Epoch 200; Iter    50/  100] train: loss: -2.2500036
[Epoch 200; Iter    52/  100] train: loss: -2.2324305
[Epoch 200; Iter    54/  100] train: loss: -2.2250788
[Epoch 200; Iter    56/  100] train: loss: -2.2158983
[Epoch 200; Iter    58/  100] train: loss: -2.2318683
[Epoch 200; Iter    60/  100] train: loss: -2.2518740
[Epoch 200; Iter    62/  100] train: loss: -2.2250538
[Epoch 200; Iter    64/  100] train: loss: -2.2269733
[Epoch 200; Iter    66/  100] train: loss: -2.2310219
[Epoch 200; Iter    68/  100] train: loss: -2.2161174
[Epoch 200; Iter    70/  100] train: loss: -2.2205598
[Epoch 200; Iter    72/  100] train: loss: -2.2224882
[Epoch 200; Iter    74/  100] train: loss: -2.2406216
[Epoch 200; Iter    76/  100] train: loss: -2.1987047
[Epoch 200; Iter    78/  100] train: loss: -2.2273459
[Epoch 200; Iter    80/  100] train: loss: -2.2188897
[Epoch 200; Iter    82/  100] train: loss: -2.2317674
[Epoch 200; Iter    84/  100] train: loss: -2.2398126
[Epoch 200; Iter    86/  100] train: loss: -2.2111089
[Epoch 200; Iter    88/  100] train: loss: -2.2469833
[Epoch 200; Iter    90/  100] train: loss: -2.2543628
[Epoch 200; Iter    92/  100] train: loss: -2.2504985
[Epoch 200; Iter    94/  100] train: loss: -2.1923199
[Epoch 200; Iter    96/  100] train: loss: -2.2006516
[Epoch 200; Iter    98/  100] train: loss: -2.2204037
[Epoch 200; Iter   100/  100] train: loss: -2.2411747
[Epoch 200] NTXent: -1.799788 val loss: -1.799788
Early stopping criterion based on -NTXent- that should be min reached after 200 epochs. Best model checkpoint was in epoch 165.
Statistics on  val_best_checkpoint
positive_similarity: 0.9688039703501595
negative_similarity: 0.49956806831889683
contrastive_accuracy: 0.759679623776012
true_negative_rate: 0.5193592525190778
true_positive_rate: 1.0
uniformity: -4.5691987540986805
alignment: 2.764178216457367
batch_variance: 0.11491788489123185
dimension_covariance: 0.0008367771305428403
NTXent: -1.797559466626909
mean_pred: -0.0002646076123306153
std_pred: 0.10940126019219558
mean_targets: 1.0975488838956457e-05
std_targets: 0.006032964382838044
Statistics on  test
positive_similarity: 0.9683920939763387
negative_similarity: 0.499461951079192
contrastive_accuracy: 0.7599385954715587
true_negative_rate: 0.519951272893835
true_positive_rate: 0.9999259268796002
uniformity: -4.5703361829121905
alignment: 2.7638528523621737
batch_variance: 0.11491100241740544
dimension_covariance: 0.0008471731475933835
NTXent: -1.8462339418905753
mean_pred: -0.0003244010140743167
std_pred: 0.10939329227915516
mean_targets: 5.363229907591934e-06
std_targets: 0.006017989599732337`
I just clone the code from the repitory without modification.
can someone tell me where the code mistake is?

**Comment by coderabbittank** [comment]
Hi，I had the same problem as you. Did you solve it?

**Comment by coderabbittank** [comment]
Hello, may I ask whether your problem has been solved now? I also have the same problem

**Comment by HannesStark** [comment]
Hi, this is because the loss is slightly different from NTXent in that it does not have the positive sample in the denominator:
<img width="470" alt="image" src="https://github.com/HannesStark/3DInfomax/assets/32128654/4403914a-7594-4b65-9792-c62a9814c543">

If you want a "correct" loss that is greater or equal to 0 you can include it.

**Comment by coderabbittank** [comment]
Thank you for your answer.

**Comment by zhihuiY** [comment]
I get it! Thank you so much!
